### PR TITLE
chore(ci): use github app for tokens

### PR DESCRIPTION
**Pull request by nirinchev**
## Description
This is a minor cleanup of gha workflows. It includes the following changes:
- Bump 3rd party actions to the latest version
- Use our github app token instead of the service account
- Cleanup redundant steps or arguments

TODO:
- [x] Update the reference to the setup-bot-token action once https://github.com/mongodb-js/devtools-shared/pull/491 is merged.

## Motivation and Context
- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [x] Misc

## Types of changes

- [ ] *Backport Needed*
- [x] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: .github/workflows/authors-and-third-party-notices.yaml [hunk 1]
```diff
@@ -16,18 +16,12 @@ jobs:
...
     steps:
       - name: Create Github App Token
-        uses: actions/create-github-app-token@v1
+        uses: mongodb-js/devtools-shared/actions/setup-bot-token@ni/github-app-action
         id: app-token
         with:
           app-id: ${{ vars.DEVTOOLS_BOT_APP_ID }}
           private-key: ${{ secrets.DEVTOOLS_BOT_PRIVATE_KEY }}
 
-      - name: Get GitHub App User ID
```
Comment (nirinchev) [inline .github/workflows/authors-and-third-party-notices.yaml:25]: The removed steps in this file have been moved to our `setup-bot-token` action.

File: .github/workflows/bump-packages.yaml [hunk 2]
```diff
@@ -10,24 +10,22 @@ jobs:
...
+          private-key: ${{ secrets.DEVTOOLS_BOT_PRIVATE_KEY }}
+
+      - uses: actions/checkout@v4
         with:
           # don't checkout a detatched HEAD
           ref: ${{ github.head_ref }}
 
-          # this is important so git log can pick up on
-          # the whole history to generate the list of AUTHORS
-          fetch-depth: '0'
```
Comment (nirinchev) [inline .github/workflows/bump-packages.yaml:20]: We're not updating AUTHORS in this workflow, so this seems unnecessary

File: .github/workflows/merge-bump-packages-pr.yaml [hunk 3]
```diff
@@ -3,18 +3,23 @@ on:
...
     # Each Tuesday at 5 AM UTC
-    - cron: '0 5 * * 2'
+    - cron: "0 5 * * 2"
 
 jobs:
   merge_bump_packages_pr:
     name: Merge bump packages PR
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
```
Comment (nirinchev) [inline .github/workflows/merge-bump-packages-pr.yaml:13]: Checkout seems unnecessary here as we don't use anything from the repo

File: .github/workflows/update-electron.yaml [hunk 4]
```diff
@@ -11,44 +11,40 @@ jobs:
...
-        run: |
-          npm -v
-          npm ci
+        run: npm ci
+
       - name: Bump packages
-        run: |
-          node scripts/update-electron.js
-          git add .
-          git commit --no-allow-empty -m "chore(deps): update electron" || true
```
Comment (nirinchev) [inline .github/workflows/update-electron.yaml:43]: The create-pull-request action will automatically add and commit, so no need to do it here

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -97,7 +97,7 @@ In particular each change to the `main` branch is analyzed to calculate a new ve
 
 Merging that PR will trigger another CI job that will publish to NPM any package which version is not yet present on the registry.
 
-The version of packages is calculated following conventional bumps: See https://github.com/mongodb-js/devtools-shared/tree/main/packages/bump-monorepo-packages for details.
+The version of packages is calculated following conventional bumps: See https://github.com/mongodb-js/devtools-shared/tree/main/packages/monorepo-tools for details.
```
Comment (nirinchev) [inline CONTRIBUTING.md:100]: This is a drive-by - noticed the link is outdated

File: .github/workflows/bump-packages.yaml [hunk 6]
```diff
@@ -40,21 +38,20 @@ jobs:
...
+          SKIP_BUMP_PACKAGES: "mongodb-compass"
         run: |
           npm run bump-packages
           git add .
           git commit --no-allow-empty -m "$LAST_BUMP_COMMIT_MESSAGE" || true
 
       - name: Create Pull Request
-        id: cpr
-        uses: peter-evans/create-pull-request@v6
+        uses: peter-evans/create-pull-request@5e914681df9dc83aa4e4905692ca88beb2f9e91f # 7.0.5
```
Comment (nirinchev) [inline .github/workflows/bump-packages.yaml:49]: For 3rd party packages, we should use git commit shas instead of version tags as those can be force pushed.

File: .github/workflows/merge-bump-packages-pr.yaml [hunk 7]
```diff
@@ -3,18 +3,23 @@ on:
...
+      - name: Create Github App Token
+        uses: actions/create-github-app-token@v1
```
Comment (gribnoysup) [inline .github/workflows/merge-bump-packages-pr.yaml:14]: Other create github app token actions are using `mongodb-js/devtools-shared/actions/setup-bot-token@ni/github-app-action`, is there a reason why this one is an exception? Should we document it?

File: .github/workflows/bump-packages.yaml (hunk 6 above)
Comment (gribnoysup) [reply .github/workflows/bump-packages.yaml:49]: Is there a documentation for this that we can point to somewhere around this code? It's not obvious and would be very easy to revert back over time without noting why we are doing this

File: .github/workflows/merge-bump-packages-pr.yaml (hunk 7 above)
Comment (nirinchev) [reply .github/workflows/merge-bump-packages-pr.yaml:14]: This one doesn't actually make commits/push. It's fairly inconsequential though and if you think it'll be less confusing we can switch it to `setup-bot-token`.

File: .github/workflows/bump-packages.yaml (hunk 6 above)
Comment (nirinchev) [reply .github/workflows/bump-packages.yaml:49]: It's part of the general security hardening recommendations for actions: https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

Comment (gribnoysup) [reply .github/workflows/bump-packages.yaml:49]: Cool, so can we add this link somewhere around this line (or any other place you think is appropriate) so that when someone is going over this code in the future and is unsure why exactly this is set to a hash instead of a version the rule is still followed?

File: .github/workflows/merge-bump-packages-pr.yaml (hunk 7 above)
Comment (gribnoysup) [reply .github/workflows/merge-bump-packages-pr.yaml]: Yes, I think it's confusing that actions with exactly the same names are using different actions, so it would be very helpful to either add a comment why or just use the same everywhere if it doesn't make a difference, maybe make it configurable to cover both cases if we don't want to use existing actions and want to have our own. New actions are often created by just using existing ones as templates, I'm not sure how one would decide which version of "Create Github App Token" to use without a clarification

File: .github/workflows/bump-packages.yaml (hunk 6 above)
Comment (nirinchev) [reply .github/workflows/bump-packages.yaml:49]: I don't think we should be adding comments as the commit sha usage should be the default for any 3rd party action. I did drop a message on slack though to bring more attention to this. I've also updated CONTRIBUTING.md with a section highlighting a few of the security best practices that also links to the full doc.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -115,6 +115,14 @@ npm run create-workspace [workspace name]
 
 This will do all the initial workspace bootstrapping for you, ensuring that your package has all the standard configs set up and ready, and all the npm scripts aligned with other packages in the monorepo, which is important to get the most out of all the provided helpers in this repository (like `npm run check-changed` commands or to make sure that your tests will not immediately fail in CI because of the test timeout being too small)
 
+## Using Github Actions
```
Comment (gribnoysup) [inline CONTRIBUTING.md:118]: Nice writeup, thanks

**Review by gribnoysup — APPROVED** [review]
